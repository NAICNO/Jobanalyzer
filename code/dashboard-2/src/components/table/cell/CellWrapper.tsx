import { ReactNode } from 'react'
import { Box, JsxStyleProps } from '@chakra-ui/react'

interface CellWrapperProps {
  styleProps?: JsxStyleProps
  children: ReactNode
}

export const CellWrapper = ({styleProps, children}: CellWrapperProps) => {
  return (
    <Box padding={1} {...styleProps}>
      {children}
    </Box>
  )
}

