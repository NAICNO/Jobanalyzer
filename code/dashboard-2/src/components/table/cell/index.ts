export { default as CellWithLink } from './CellWithLink'
export { default as CellWrapper } from './CellWrapper'
export { default as GenericCell } from './GenericCell'
export { default as GpuFieldCell } from './GpuFieldCell'
export { default as WorkingFieldCell } from './WorkingFieldCell'
