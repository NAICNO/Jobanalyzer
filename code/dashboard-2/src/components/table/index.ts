export * from './DasboardTable'
export * from './DashboardTableRow'
export * from './DeadWeightTable'
export * from './JobQueryResultsTable'
export * from './TableHeader'
export * from './TableRow'
export * from './ViolatingJobTable'
export * from './ViolatingUserTable'
