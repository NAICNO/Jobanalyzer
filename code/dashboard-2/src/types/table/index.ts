export * from './DashboardTableColumnHeader.ts'
export * from './DashboardTableItem.ts'
export * from './DeadWeightTableColumnHeader.ts'
export * from './DeadWeightTableItem.ts'
export * from './JobQueryJobId.ts'
export * from './JobQueryResultsTableColumnHeader.ts'
export * from './JobQueryResultsTableItem.ts'
export * from './ViolatingJobTableColumnHeader.ts'
export * from './ViolatingJobTableItem.ts'
export * from './ViolatingUserTableColumnHeader.ts'
export * from './ViolatingUserTableItem.ts'
