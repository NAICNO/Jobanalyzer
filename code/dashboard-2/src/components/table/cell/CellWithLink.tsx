import { Link as ReactRouterLink } from 'react-router-dom'
import { Link as ChakraLink } from '@chakra-ui/react'

import CellWrapper from './CellWrapper.tsx'

interface CellWithLinkProps {
  value: TextWithLink
}

const CellWithLink = ({value}: CellWithLinkProps) => {
  return (
    <CellWrapper styleProps={{paddingLeft: 4}}>
      <ChakraLink as={ReactRouterLink} to={value.link}>
        {value.text}
      </ChakraLink>
    </CellWrapper>
  )
}

export default CellWithLink
