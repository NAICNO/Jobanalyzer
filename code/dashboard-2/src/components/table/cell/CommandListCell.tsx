import { Text } from '@chakra-ui/react'

import { CellWrapper } from './CellWrapper.tsx'
import { breakText } from '../../../util'

interface CommandListCellProps {
  value: string;
}

export const CommandListCell = ({value}: CommandListCellProps) => {
  const brokenText = breakText(value)

  return (
    <CellWrapper>
      <Text>{brokenText}</Text>
    </CellWrapper>
  )
}
