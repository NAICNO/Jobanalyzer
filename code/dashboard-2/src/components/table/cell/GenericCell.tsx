import { Text } from '@chakra-ui/react'
import CellWrapper from './CellWrapper.tsx'

interface GenericCellProps {
  value: number;
}

const GenericCell = ({value}: GenericCellProps) => {
  return (
    <CellWrapper>
      <Text>{value}</Text>
    </CellWrapper>
  )
}

export default GenericCell
