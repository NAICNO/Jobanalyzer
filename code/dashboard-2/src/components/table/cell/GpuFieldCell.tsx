import { Text } from '@chakra-ui/react'

import { CELL_BACKGROUND_COLORS } from '../../../Constants.ts'
import { CellWrapper } from './CellWrapper.tsx'
import { useColorModeValue } from '../../ui/color-mode.tsx'

interface GpuFieldCellProps {
  value: number;
}

export const GpuFieldCell = ({value}: GpuFieldCellProps) => {

  const isGpuDownOrUndefined = value !== 0 || value === undefined

  const backgroundColor = useColorModeValue(
    isGpuDownOrUndefined ? CELL_BACKGROUND_COLORS.LIGHT.DOWN : CELL_BACKGROUND_COLORS.LIGHT.NA,
    isGpuDownOrUndefined ? CELL_BACKGROUND_COLORS.DARK.DOWN : CELL_BACKGROUND_COLORS.DARK.NA
  )

  return (
    <CellWrapper styleProps={{backgroundColor}}>
      <Text>{value}</Text>
    </CellWrapper>
  )
}
