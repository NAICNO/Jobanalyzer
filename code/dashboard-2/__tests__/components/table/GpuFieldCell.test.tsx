import '@testing-library/jest-dom'
import { render } from '../../test-utils.tsx'

import { GpuFieldCell } from '../../../src/components/table/cell'

describe('GpuFieldCell', () => {
  it('renders correctly when value is not 0 or undefined', () => {
    const value = 1
    const {asFragment} = render(<GpuFieldCell value={value}/>)
    expect(asFragment()).toMatchSnapshot()
  })

  it('renders correctly when value is 0', () => {
    const value = 0
    const {asFragment} = render(<GpuFieldCell value={value}/>)
    expect(asFragment()).toMatchSnapshot()
  })
})
