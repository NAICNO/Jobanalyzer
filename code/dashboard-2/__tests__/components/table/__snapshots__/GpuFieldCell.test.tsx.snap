// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`GpuFieldCell > renders correctly when value is 0 1`] = `
<DocumentFragment>
  <script
    nonce=""
  >
    ((e,i,s,u,m,a,l,h)=&gt;{let d=document.documentElement,w=["light","dark"];function p(n){(Array.isArray(e)?e:[e]).forEach(y=&gt;{let k=y==="class",S=k&&a?m.map(f=&gt;a[f]||f):m;k?(d.classList.remove(...S),d.classList.add(a&&a[n]?a[n]:n)):d.setAttribute(y,n)}),R(n)}function R(n){h&&w.includes(n)&&(d.style.colorScheme=n)}function c(){return window.matchMedia("(prefers-color-scheme: dark)").matches?"dark":"light"}if(u)p(u);else try{let n=localStorage.getItem(i)||s,y=l&&n==="system"?c():n;p(y)}catch(n){}})("class","theme","system",null,["light","dark"],null,true,true)
  </script>
  <div
    class="css-xrjgq3"
  >
    <p
      class="css-0"
    >
      0
    </p>
  </div>
</DocumentFragment>
`;

exports[`GpuFieldCell > renders correctly when value is not 0 or undefined 1`] = `
<DocumentFragment>
  <script
    nonce=""
  >
    ((e,i,s,u,m,a,l,h)=&gt;{let d=document.documentElement,w=["light","dark"];function p(n){(Array.isArray(e)?e:[e]).forEach(y=&gt;{let k=y==="class",S=k&&a?m.map(f=&gt;a[f]||f):m;k?(d.classList.remove(...S),d.classList.add(a&&a[n]?a[n]:n)):d.setAttribute(y,n)}),R(n)}function R(n){h&&w.includes(n)&&(d.style.colorScheme=n)}function c(){return window.matchMedia("(prefers-color-scheme: dark)").matches?"dark":"light"}if(u)p(u);else try{let n=localStorage.getItem(i)||s,y=l&&n==="system"?c():n;p(y)}catch(n){}})("class","theme","system",null,["light","dark"],null,true,true)
  </script>
  <div
    class="css-16upyym"
  >
    <p
      class="css-0"
    >
      1
    </p>
  </div>
</DocumentFragment>
`;
